src/resmom/mom_hook_func.c

```diff
@@ -3192,6 +3474,28 @@ mom_process_hooks(unsigned int hook_event, char *req_user, char *req_host,
 		default:
 			return (-1); /* unexpected event encountered */
 	}
+#ifndef WIN32
+	if (hook_event == HOOK_EVENT_EXECJOB_END) {
+		/* Unblock MOM on execjob_end event */
+
+		struct 			work_task task;
+		task.wt_parm1 = (void *)hook_input;
+		task.wt_parm2 = NULL;
+		if (!run_execjob_end_hooks(&task)) {
+			/* on an execjob_end, need to set the job's exit value.     */
+			/* Set it only once, and only if there's a hook to execute  */
+			/* since we're affecting the job directly.                  */
+
+			pjob->ji_wattr[(int)JOB_ATR_exit_status].at_val.at_long = \
+				pjob->ji_qs.ji_un.ji_momt.ji_exitstat;
+			pjob->ji_wattr[(int)JOB_ATR_exit_status].at_flags |= (ATR_VFLAG_SET | ATR_VFLAG_MODCACHE);
+
+			pjob->ji_execjob_end_hook_event_started = 1;
```

Comment:
I see pjob->ji_execjob_end_hook_event_started set to 1, but where in the code does this ever get set to 0?  Also, there should be an initialization entry for this field in src/server/job_func.c:job_alloc().

Reply:
Thanks for the review Al, 
Done, I initialized in job_alloc (),  a reset is not required as the job will get purged after this event.

Reply:
it would be prudent to reset this field once all the execjob_end hooks are executed.